checker.cpp

```diff
@@ -1,22 +1,41 @@
 #include <assert.h>
 #include <iostream>
+#include "checker.hpp"
 using namespace std;
-
-bool batteryIsOk(float temperature, float soc, float chargeRate) {
-  if(temperature < 0 || temperature > 45) {
+using namespace BatteryStatus;
+bool Battery::batteryIsOk(std::vector<float>BatteryFunctionParameters) {
```

Comment:
batteryIsOk() seems to be just a delegate method to execute a vector of function pointers.
Whereas, from the name, it seems like a method that knows exactly what all parameters to check within itself.

Reply:
ok.